### PR TITLE
 [OCPBUGS-23115] Builds updates customer case

### PR DESCRIPTION
**Pull request by nebelsprachen**
Version(s): 

Issue:

* [OCPBUGS-23115](https://issues.redhat.com/browse/OCPBUGS-23115)

Link to docs preview:

* https://68699--docspreview.netlify.app/openshift-enterprise/latest/cicd/builds/running-entitled-builds

QE review:
- [ ] QE has approved this change.

Additional information:

* Urgent customer case
* Related to this engineering issue: [OCPBUGS-23285](https://issues.redhat.com/browse/OCPBUGS-23285)

### CODE REVIEW

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc [hunk 1]
```diff
@@ -61,15 +62,47 @@ rules:
...
 [source,terminal]
 ----
 $ oc create rolebinding shared-resource-my-share --role=shared-resource-my-share --serviceaccount=my-namespace:builder
 ----
++
+. Add a volume to mount the `rhsm.conf` file into the container. Specify the path to `rhsm.conf` as the source for this volume.
++
+[TIP]
+=== 
+You must have access to Red Hat entitlements through your subscription. The entitlement secret is automatically created by the Insights Operator. When you perform an Entitlement Build using Red Hat Enterprise Linux (RHEL) 7, you must have the following instructions in your Dockerfile before you run any yum commands:
```
Comment (adambkaplan) [inline modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc]: Please check the terminology - my understanding is that the cluster is _subscribed_. Once the cluster is subscribed, the entitlements are granted and synced via the Insights Operator.

```suggestion
Your cluster must be attached to an active Red Hat subscription. The entitlement secret is automatically created by the Insights Operator.

When using RHEL 7 or UBI8 as the base image of your build, you must add the following instructions to your Dockerfile before executing any `yum` or `dnf` commands:
```

**Review by adambkaplan — CHANGES_REQUESTED** [review]
I don't want to rush the doc changes here. As we have learned, our instructions have skew between UBI/RHEL 7, 8, and 9. We need to provide separate Dockerfile/BuildConfig instructions and examples for each version, since these are all supported versions of RHEL and are compatible with OpenShift.

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc [hunk 2]
```diff
@@ -4,24 +4,20 @@
...
-====
+The Shared Resources CSI Driver feature also belongs to the `TechPreviewNoUpgrade` feature set, which is a subset of the current Technology Preview features. You can enable the `TechPreviewNoUpgrade` feature set on test clusters, where you can fully test them while leaving the features disabled on production clusters. Enabling this feature set cannot be undone and prevents minor version updates. This feature set is not recommended on production clusters. See "Enabling Technology Preview features using feature gates" in the following "Additional resources" section.
 
 .Prerequisites
 
-* You have enabled the  `TechPreviewNoUpgrade` feature set by using the feature gates.
+* You have enabled the `TechPreviewNoUpgrade` feature set by using the feature gates.
 * You have a `SharedSecret` custom resource (CR) instance that references the `Secret` object where the Insights Operator stores the subscription credentials.
+* You must have access to Red Hat entitlements through your subscription.
```
[the remark this answers is not in the document]
Comment (control-d) [reply modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc:20]: If the cluster is Connected (ie, not in a disconnected network environment) and the Insights Operator was not excluded from the cluster install, then I believe this happens automatically. 

My understanding is that the entitlement will be associated with the Red Hat account tied to the pull secret used to install the cluster.

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc [hunk 3]
```diff
@@ -78,35 +111,41 @@ $ oc create rolebinding shared-resource-my-share --role=shared-resource-my-share
...
-            csi:
-              driver: csi.sharedresource.openshift.io
-              readOnly: true
-              volumeAttributes:
-                sharedSecret: my-share-bc
-            type: CSI
+            secret:
+              defaultMode: 420
+              secretName: etc-pki-entitlement
+            type: Secret
```
Comment (control-d) [inline modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc:146]: This is the sharedSecret doc, so I think we want to keep the existing CSI SharedResource example and not show the use of a standalone secret.

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc [hunk 4]
```diff
@@ -78,35 +111,41 @@ $ oc create rolebinding shared-resource-my-share --role=shared-resource-my-share
...
+          crontabs \
+          nss_wrapper \
+          uid_wrapper -y && \
+          yum clean all -y
   strategy:
     type: Docker
     dockerStrategy:
+      from:
+        kind: DockerImage
+        name: registry.access.redhat.com/ubi8/ubi:8.7-1112
```
Comment (control-d) [inline modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc:137]: I don't suggest using an example that pins to a specific image. This image, [8.7-1112](https://catalog.redhat.com/software/containers/ubi8/5c647760bed8bd28d0e38f9f?architecture=amd64&image=64226b8a5f05d70b2f432ce2&container-tabs=security), is already quite old and missing a number of security fixes. Using `ubi9/ubi:latest` is likely to be a safer, better example.

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc [hunk 5]
```diff
@@ -78,35 +111,41 @@ $ oc create rolebinding shared-resource-my-share --role=shared-resource-my-share
 apiVersion: build.openshift.io/v1
 kind: BuildConfig
 metadata:
-  name: my-csi-bc
-  namespace: my-csi-app-namespace
+  name: entitled-build
+  namespace: entitlement-test
```
[the remark this answers is not in the document]
Comment (joaedwar) [reply modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc:115]: Hi @adambkaplan. The publishing platform does not allow for separate tabs. We would have to create an entry for each.

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc (hunk 4 above)
Comment (joaedwar) [reply modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc:137]: HI @adambkaplan and @control-d . I need some help in better understanding what example we can provide, for customers. What do you think would be a good generic example to present in our docs?

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (adambkaplan) [reply modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc]: I think this is correct. @tremes I believe you lead the Insights Operator team and can confirm this?

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (adambkaplan) [reply modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc:114]: From the working group- the current recommendation is to remove that /etc/rhsm-host symbolic link as a RUN step in the build for UBI7 and UBI8 based images. @openshift/pipeline-integrations-team needs to test/verify if this is still needed for UBI9 images.

**Pull request closed without merging**